### PR TITLE
Melkehitys-1054

### PR DESCRIPTION
**Pull request by natlibfi-jonollil**

### CODE REVIEW

**Comment by natlibfi-jonollil** [comment]
Routes / Bulk
readContent, remove and removeContent are there for testing purposes

readContent: to check that saved that is if OK
remove & removeContent: to remove failed test loads

**Comment by natlibfi-jonollil** [comment]
Routes / Prio: Offlinehours
There is still down time on server for maintenance purposes? To avoid situation where request is in process and server goes to maintenance.

**Comment by natlibfi-jonollil** [comment]
Interfaces / Bulk
Queries can be dummed down later. Atm. they are here to help me check old load stats for testing

**Comment by natlibfi-arlehiko** [comment]
> There is still down time on server for maintenance purposes? To avoid situation where request is in process and server goes to maintenance.

Shouldn't  be necessary now that the record load API should return the resource id immediately and the importer just polls for it.

File: src/interfaces/prio.js [hunk 1]
```diff
@@ -138,8 +140,13 @@ export default async function ({sruBibUrl, amqpUrl, pollWaitTime}) {
 			return message;
 		}
 
+		tries++;
+		if (tries > 1200) {
```
Comment (natlibfi-arlehiko) [inline src/interfaces/prio.js]: Don't modify variables. Just do `if (tries + 1 > 1200) {`

File: src/interfaces/prio.js [hunk 2]
```diff
@@ -138,8 +140,13 @@ export default async function ({sruBibUrl, amqpUrl, pollWaitTime}) {
...
+			throw new ApiError(408);
+		}
+
 		// Nothing in queue
 		await setTimeoutPromise(pollWaitTime);
-		return check(queue);
+		return check(queue, tries);
```
Comment (natlibfi-arlehiko) [inline src/interfaces/prio.js]: Change to `tries + 1`

File: src/app.js [hunk 3]
```diff
@@ -29,9 +29,7 @@ export default async function ({
 	async function initExpress() {
 		const app = express();
 
-		if (enableProxy) {
-			app.enable('trust proxy', true);
-		}
+		proxySettings(enableProxy);
```
Comment (natlibfi-arlehiko) [inline src/app.js]: Change to `app.enable('trust proxy', enableProxy ? true : false);`

File: src/index.js [hunk 4]
```diff
@@ -52,20 +52,26 @@ async function run() {
...
-				server.close();
-			}
-
-			if (message) {
-				logError(message);
-			}
-
+		function handleTermination({code = 0, message = false}) {
+			logMessage(message);
+			closeServer();
```
Comment (natlibfi-arlehiko) [inline src/index.js]: Change to:
```js
if (server) {
  return to server.close();
}
```

File: src/app.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    app.use(bodyParser.text({limit: '5MB', type: '*/*'}));
+    app.use('/apidoc', createApiDocRouter());
+    app.use('/', await createPrioRouter({sruBibUrl, amqpUrl, pollWaitTime}));
+
+    app.use(handleError);
+
+    return app.listen(httpPort, () => logger.log('info', `Started Melinda REST API in port ${httpPort}`));
+
+    function handleError(err, req, res) {
+      // The correct way would be to throw if the error is unexpected...There is a race condition between the request aborted event handler and running async function.
```
Comment (natlibfi-arlehiko) [inline src/app.js:53]: If this the comment from the OAI-PMH provider you can remove it :) There seems to be no such special handling in this function (IIRC, the problem was with express, async request handlers + Oracle).

File: src/index.js [hunk 6]
```diff
@@ -34,38 +34,43 @@ import {logError} from '@natlibfi/melinda-rest-api-commons';
...
-	const {handleInterrupt} = Utils;
-	let server;
+  const {handleInterrupt} = Utils;
+  let server; // eslint-disable-line functional/no-let
 
-	registerInterruptionHandlers();
+  registerInterruptionHandlers();
 
-	server = await startApp({...config});
+  server = await startApp({...config}); // eslint-disable-line prefer-const
```
Comment (natlibfi-arlehiko) [inline src/index.js:42]: Do as in [this](https://github.com/NatLibFi/aleph-own-auth-api/blob/next/src/index.js).

File: src/interfaces/prio.js [hunk 7]
```diff
@@ -36,110 +36,115 @@ const setTimeoutPromise = promisify(setTimeout);
...
+  async function check(queue, tries = 0) {
+    // Check queue
+    const message = await amqpOperator.checkQueue(queue, 'raw', false);
+
+    if (message) {
+      return message;
+    }
+
+    // To close infinite loops
+    if (tries + 1 > 1200) { // eslint-disable-line functional/no-conditional-statement
```
Comment (natlibfi-arlehiko) [inline src/interfaces/prio.js:142]: What does this hardcoded value `1200` represent?

**Pull request merged**